restorer.go

```diff
@@ -4,33 +4,42 @@ import (
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 
+	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/cmd"
 	"github.com/buildpacks/lifecycle/layers"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Restorer struct {
-	LayersDir  string
-	Buildpacks []buildpack.GroupBuildpack
-	Logger     Logger
+	LayersDir              string
+	Buildpacks             []buildpack.GroupBuildpack
+	Logger                 Logger
+	SkipLayers             bool
+	LayerAnalyzer          LayerAnalyzer
+	CacheMetadataRetriever CacheMetadataRetriever
+	Platform               cmd.Platform
+	LayersMetadata         platform.LayersMetadata
```

Comment:
Could we add a comment to indicate which fields are required for which api?

Reply:
Also same comment as https://github.com/buildpacks/lifecycle/pull/582/files#r614901576 re: ordering.